rest/api_collections_test.go

```diff
@@ -155,6 +155,69 @@ func TestCollectionsPublicChannel(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, 1, alldocsresp.TotalRows)
 	assert.Len(t, alldocsresp.Rows, 1)
+
+	// user doc and accessible doc
+	changesResults := rt.WaitForChanges(2, "/{{.keyspace}}/_changes", username, false)
+	require.Len(t, changesResults.Results, 2)
+	t.Logf("changes results: %s", changesResults.Summary())
+}
+
+// TestCollectionsPublicChannelViaSyncFn ensures that docs routed to the public channel via the sync function are accessible by a user with no other access.
+func TestCollectionsPublicChannelViaSyncFn(t *testing.T) {
+	const (
+		username = "alice"
+		password = RestTesterDefaultUserPassword
+	)
+
+	rt := NewRestTester(t, &RestTesterConfig{
+		SyncFn: `function(doc){channel('!');}`,
```

Comment:
Do you think this test is different than above?

If I understand correctly, when collections are enabled, `SyncFn: channels.DocChannelsSyncFunction` does execute javascript because that is no longer the implicit sync function.